### PR TITLE
Add the Modus colorscheme

### PR DESCRIPTION
**Pull request by nwjsmith**
Add the Modus colorscheme

### CODE REVIEW

File: lib/maintainers.nix [hunk 1]
```diff
@@ -54,4 +54,11 @@
     githubId = 66408983;
     name = "svl";
   };
+  nwjsmith = {
+    email = "nate@theinternate.com";
+    github = "nwjsmith";
+    githubId = 1348;
+    name = "Nate Smith";
+    keys = [ { fingerprint = "DE6F E362 7DF7 EEA0 79CB  38C1 F208 9547 768B 6CCA"; } ];
+  };
```
Comment (MattSturgeon) [inline lib/maintainers.nix:63]: We prefer this to be done in a separate commit titled "`maintainers: add nwjsmith`"

File: plugins/colorschemes/modus.nix
Comment (MattSturgeon) [inline plugins/colorschemes/modus.nix:1]: This commit should be titled "`colorschemes/modus: init`"

File: plugins/colorschemes/modus.nix [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    styles = {
+      comments.italic = true;
+      keywords.italic = true;
+      functions = { };
+      variables = { };
+    };
+    on_colors = "function(colors) end";
+    on_highlights = "function(highlights, colors) end";
+  };
+}
```
Comment (MattSturgeon) [inline plugins/colorschemes/modus.nix:93]: Most colorschemes set `opts.termguicolors = mkDefault true` in `extraConfig`. Does this plugin need `termguicolors`?

**Comment by nwjsmith** [comment]
@MattSturgeon thanks for the quick review. Addressed all of your feedback.

File: plugins/colorschemes/modus.nix (hunk 2 above)
Comment (nwjsmith) [reply plugins/colorschemes/modus.nix:95]: Good catch, it is required. It's based on tokyonight which also requires it. Fixed.

File: plugins/colorschemes/modus.nix [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+      comments.italic = true;
+      keywords.italic = true;
+      functions = { };
+      variables = { };
+    };
+    on_colors = "function(colors) end";
+    on_highlights = "function(highlights, colors) end";
+  };
+
+  extraConfig = cfg: { opts.termguicolors = lib.mkDefault true; };
```
Comment (MattSturgeon) [inline plugins/colorschemes/modus.nix:94]: I can't find any mention of `termguicolors` on the upstream readme, so maybe it isn't needed for this plugin. Someone more knowledgeable than me should probably weigh in here.

File: plugins/colorschemes/modus.nix [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+      Override specific color groups to use other groups or a hex color.
+      Function will be called with a `ColorScheme` table.
+      `@param colors ColorScheme`
+    '';
+
+    on_highlights = helpers.defaultNullOpts.mkLuaFn "function(highlights, colors) end" ''
+      Override specific highlights to use other groups or a hex color.
+      Function will be called with a `Highlights` and `ColorScheme` table.
+      `@param highlights Highlights`
+      `@param colors ColorScheme`
```
Comment (MattSturgeon) [inline plugins/colorschemes/modus.nix:75]:
```suggestion
      Override specific highlights to use other groups or a hex color.
      Function will be called with a `Highlights` and `ColorScheme` table.

      ```
        @param highlights Highlights
        @param colors ColorScheme
      ```
```

File: plugins/colorschemes/modus.nix [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+
+      variables = helpers.defaultNullOpts.mkHighlight "{}" "" ''
+        Define variables highlight properties.
+      '';
+    };
+
+    on_colors = helpers.defaultNullOpts.mkLuaFn "function(colors) end" ''
+      Override specific color groups to use other groups or a hex color.
+      Function will be called with a `ColorScheme` table.
+      `@param colors ColorScheme`
```
Comment (MattSturgeon) [inline plugins/colorschemes/modus.nix:68]:
```suggestion
      Override specific color groups to use other groups or a hex color.
      Function will be called with a `ColorScheme` table.

      ```
        @param colors ColorScheme
      ```
```

File: plugins/colorschemes/modus.nix [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+
+  settingsOptions = {
+    style =
+      helpers.defaultNullOpts.mkEnumFirstDefault
+        [
+          "modus_operandi"
+          "modus_vivendi"
+        ]
+        ''
+          The theme comes in two styles, `modus_operandi`, a dark variant `modus_vivendi`.
```
Comment (MattSturgeon) [inline plugins/colorschemes/modus.nix:24]: It feels a bit redundant to list the available styles here when the mkEnum function will do that anyway...

Maybe this and `variant`'s description should instead try to describe what the option actually does and/or what things it affects?

Doesn't look like there's anything to quote in upstream's readme though, so I appreciate if you don't want to do this.

**Review by MattSturgeon — COMMENTED** [review]
Looking good to me, just some minor comments/suggestions

**Comment by MattSturgeon** [comment]
[Looks like](https://buildbot.nix-community.org/api/v2/logs/58322/raw_inline) something is going wrong when attempting to load modus from the runtimepath?

File: plugins/colorschemes/modus.nix [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+  helpers,
+  config,
+  pkgs,
+  ...
+}:
+helpers.neovim-plugin.mkNeovimPlugin config {
+  name = "modus";
+  isColorscheme = true;
+  originalName = "modus-themes-nvim";
+  defaultPackage = pkgs.vimPlugins.modus-themes-nvim;
```
Comment (MattSturgeon) [inline plugins/colorschemes/modus.nix:12]: Add `luaName = "modus-themes"` to fix requiring the plugin.

Personally, I'd re-order these as follows, but OTTOMH IDK if that's consistent with what we do in other plugins:


```suggestion
  name = "modus";
  luaName = "modus-themes";
  originalName = "modus-themes-nvim";
  defaultPackage = pkgs.vimPlugins.modus-themes-nvim;
  isColorscheme = true;
```

File: plugins/colorschemes/modus.nix (hunk 6 above)
Comment (nwjsmith) [reply plugins/colorschemes/modus.nix]: Good call. I've cribbed new descriptions [from documentation on the original Emacs themes](https://protesilaos.com/emacs/modus-themes#h:f0f3dbcb-602d-40cf-b918-8f929c441baf).

File: plugins/colorschemes/modus.nix (hunk 3 above)
Comment (nwjsmith) [reply plugins/colorschemes/modus.nix:98]: The theme is based on the Tokyo Night, and the colors [are specified as 24-bit RGB values](https://github.com/miikanissi/modus-themes.nvim/blob/master/lua/modus-themes/colors.lua) so I believe `termguicolors` is required.

**Review by MattSturgeon — CHANGES_REQUESTED** [review]
A bit of further feedback on the new variant description. Otherwise looks good.

Any additional thoughts @GaetanLepage?

File: plugins/colorschemes/modus.nix [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+          "default"
+          "tinted"
+          "deuteranopia"
+          "tritanopia"
+        ]
+        ''
+          Styles come in four variants. `default` is the project's main style designed to cover a
+          broad range of needs. `tinted` tone down intensity and provide more color variety.
+          `deuteranopia` are optimized for users with red-green color deficiency. `tritanopia` are
+          optimized for users with blue-yellow color deficiency.
```
[the remark this answers is not in the document]
Comment (nwjsmith) [reply plugins/colorschemes/modus.nix]: I like it. Switched to a bulleted list and fixed the grammar. Thank you!

File: plugins/colorschemes/modus.nix [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+  lib,
+  helpers,
+  config,
+  pkgs,
+  ...
+}:
+helpers.neovim-plugin.mkNeovimPlugin config {
+  name = "modus";
+  luaName = "modus-themes";
+  originalName = "modus-themes-nvim";
```
Comment (MattSturgeon) [inline plugins/colorschemes/modus.nix:11]: My bad for not spotting this sooner:

```suggestion
  originalName = "modus-themes.nvim";
```

**Review by MattSturgeon — APPROVED** [review]
Thanks for your efforts and perseverance!
